src/Shared/Recruit.Vacancies.Client/Infrastructure/Client/VacancyClient.cs

```diff
@@ -387,6 +387,16 @@ public Task WithdrawApplicationAsync(long vacancyReference, Guid candidateId)
             });
         }
 
+        public async Task EnsureVacancyHasReferenceNumberAsync(Guid vacancyId)
+        {
+            var vacancy = await _repository.GetVacancyAsync(vacancyId);
```

Comment:
Do we need to retrieve the vacancy? Is this not just duplicate logic of what is in the command handler?

Reply:
This is for the purpose of the check to decide whether it needs to be assigned.

Reply:
Isn't it just duplicating what's in the handler?

https://github.com/SkillsFundingAgency/das-recruit/blob/0e7546ee7d93ff7bd53aa0d811fd311744deb685/src/Shared/Recruit.Vacancies.Client/Application/CommandHandlers/AssignVacancyNumberCommandHandler.cs#L36-L41